Dang 193/consistent input states

mercedesb opened the pull request:
## JIRA

* https://lobsters.atlassian.net/browse/DANG-193

<!--
## Technical Design Doc
- link to TDD if the feature had one (optional)
-->

## Description

* Make all of our input states consistent
* Inputs should have `required`, `error`, `disabled`, and `readonly` props (except for the dropdown which does not have `readonly` because it is not supported in the native HTML select)
* When disabled or readonly, inputs should be `bg-white-300` and `cursor-not-allowed`
* When disabled or readonly, there should be no hover shadows
* The focus states should be consistent, `primary-100` and ring instead of outline
* Placeholders should be styled consistently

<!--
## Screenshots
- before and after screenshots for features with visual changes
-->

<!--
## Tests
- info about what was used to validate the feature (more for larger, more complicated features)
-->

<!--
## Areas of Concern
- to call out what could be a problem (optional)
-->

<!--
## Questions
- to ask any questions for the reviewers (optional)
-->

<!--
## GIFs/Memes
- (optional)
-->

## Reviewer Checklist

This section is to be filled out by reviewers

### Testing

- [ ] This code was tested by somebody other than the developer. Do not merge until this has been done.

juanfriss reviewed (changes requested):
Checkbox:
- When the checkbox is readonly, it's actually allowing clicks and changing its value
- Not sure if it's clear/readable when the checkbox is checked but disabled/readonly. Too little contrast between the checkmark and the background.

Text input and text area:
- They currently do nothing when required. Maybe add a little * to the label so the user knows its required, as in the checkbox and radio button group?

mercedesb commented:
> Checkbox:
> 
> * When the checkbox is readonly, it's actually allowing clicks and changing its value
> * Not sure if it's clear/readable when the checkbox is checked but disabled/readonly. Too little contrast between the checkmark and the background.
> 
> Text input and text area:
> 
> * They currently do nothing when required. Maybe add a little * to the label so the user knows its required, as in the checkbox and radio button group?

@juanfriss 

- Oops, readonly isn't supported on input[type=checkbox] because readonly is to prevent a _value_ from changing, not the _state_. I went ahead and removed that.
- Updated contrast, was missing a `!`
![image](https://user-images.githubusercontent.com/9774690/123119979-96849e80-d409-11eb-982b-f969950cefea.png)
- Added asterisk
![image](https://user-images.githubusercontent.com/9774690/123120171-b7e58a80-d409-11eb-8f2c-31ec399523dd.png)
